data/data.rb

```diff
@@ -0,0 +1,169 @@
+    myfile.write(JSON.pretty_generate(book))
+    myfile.close
+  end
+end
+
+def load_people
+  if File.exist?('./data/people.json')
+    file = File.open('./data/people.json')
+    if File.empty?('./data/people.json')
+      puts 'Please add people data if this is your first time visiting our app'
+    else
+      people = JSON.parse(File.read('./data/people.json'))
+      people.each do |person|
+        if person['option'] == 'Student'
+          student = Student.new(person['age'], person['name'])
+          @people << student
+        else
+          teacher = Teacher.new(person['specialization'], person['age'], person['name'])
+          @people << teacher
+        end
+      end
+    end
+    file.close
+  else
+    puts 'Please insert some data'
+  end
+end
+
+def save_student(age, name, parent_permission)
+  obj = {
+    type: 'Student',
+    age: age,
+    name: name,
+    parent_permission: parent_permission
+  }
+
+  if File.exist?('./data/people.json')
+    file = File.open('./data/people.json')
+
+    if File.empty?('./data/people.json')
+      student = [obj]
+    else
+      student = JSON.parse(File.read('./data/people.json'))
+      student << obj
+    end
+
+    file.close
+
+    myfile = File.open('./data/people.json', 'w')
+    myfile.write(JSON.pretty_generate(student))
+    myfile.close
+  end
+end
+
+def save_teacher(specialization, age, name)
+  obj = {
+    type: 'Teacher',
+    specialization: specialization,
+    name: name,
+    age: age,
+    parent_permission: true
+  }
+
+  if File.exist?('./data/people.json')
+    file = File.open('./data/people.json')
+
+    if File.empty?('./data/people.json')
+      teacher = [obj]
+    else
+      teacher = JSON.parse(File.read('./data/people.json'))
+      teacher << obj
+    end
+
+    file.close
+
+    myfile = File.open('./data/people.json', 'w')
+    myfile.write(JSON.pretty_generate(teacher))
+    myfile.close
+  end
+end
+
+def load_rentals
+  if File.exist?('./data/rentals.json')
+    file = File.open('./data/rentals.json')
+
+    if File.empty?('./data/rentals.json')
+      puts 'Please add some books first'
+    else
+      rentals = JSON.parse(File.read('./data/rentals.json'))
+      puts 'Reserved books: '
+      rentals.each do |rental|
+        puts "Name: #{rental['person']}, Book: #{rental['book']} on: #{rental['date']}"
+      end
+    end
+    file.close
+  else
+    puts 'No reservations found, please reseve some books first'
+  end
+end
+
+def save_rentals(date, book, person)
+  obj = {
+    date: date,
+    person: person.name,
+    book: book.title
+  }
+
+  if File.exist?('./data/rentals.json')
+    file = File.open('./data/rentals.json')
+
+    if File.empty?('./data/rentals.json')
+      rental = [obj]
+    else
+      rental = JSON.parse(File.read('./data/rentals.json'))
+      rental << obj
+    end
+
+    file.close
+
+    myfile = File.open('./data/rentals.json', 'w')
+    myfile.write(JSON.pretty_generate(rental))
+    myfile.close
+  end
```

Comment:
- [ ] One major requirement for this milestone is to ensure that the application doesn't crash and functions well if the JSON files used to save data are missing. Although the application doesn't crash, the data are lost upon exit of the application when the JSON files are missing or deleted. I would advise that you consider this factor in your code when accessing the file. Please implement this for the book, person and rental JSON files. 💻 👍 

#### Hint: 

```ruby
  File.new('./data/filename.json', 'w') unless File.exist?('./data/filename.json')
```